src/OrleansProviders/Streams/Common/PooledCache/ObjectPool.cs

```diff
@@ -14,7 +15,7 @@ public class ObjectPool<T> : IObjectPool<T>
         where T : PooledResource<T>
     {
         private const int DefaultPoolCapacity = 1 << 10; // 1k
-        private readonly Stack<T> pool;
+        private readonly ConcurrentStack<T> pool;
```

Comment:
Change in this file is to make ObjectPool to be able to handle concurrency when allocate and free blocks